Starter: Main menu sketch + default word deserialization

Z1Epime opened the pull request:
- Added first sketch of the main menu
- Added model "Word"
- Added JSON deserialization structure of default words
- Added NuGet Packages:
  - CommunityToolKit for MVVM
  - Microsoft.Extensions.DependencyInjection
  - Microsoft.Xaml.Behaviors.WPF
  - Newtonsoft.Json

Z1Epime committed:
Getting started

- Added model "Word"
- Added JSON serialization/deserialization structure
- Added NuGet Packages:
  - CommunityToolKit for MVVM
  - Microsoft.Extensions.DependencyInjection
  - Microsoft.Xaml.Behaviors.WPF
  - Newtonsoft.Json

MibFromGit commented on DataAccess/Model/Word.cs:
```diff
@@ -0,0 +1,34 @@
+        /// <summary>
+        /// The letters of a word in their correct order.
+        /// </summary>
+        public string Letters { get; set; }
```
hier können wir evtl nen Char array draus machen da es eine string.ToCharArray Methode gibt die einen Char Array erstellt und wir so das ganze nur einmal machen müssen und die Funktion nicht immer aufrufen müssen

FabioFlunder replied:
Since we can call String[Index] this isn't neccessary. The council has deemed your proposal superfluous.

MibFromGit commented on DataAccess/JSON/WordAccessor.cs:
```diff
@@ -0,0 +1,60 @@
+        /// in the ApplicationData folder.
+        /// </summary>
+        /// <param name="words">The words to serialize in the hardcoded path.</param>
+        public async Task SetWords(List<Word> words)
```
aktuell muss diese Methode nicht public sein und wird sie sehr wahrscheinlich auch nie, daher können wir sie auf private setzen und in eine Property Laden welche wir dann in der GetWords methode auf null prüfen und auf bedarf Laden sollte es vorher noch nicht getan worden sein...

Z1Epime replied:
Was resolved with new structural changes; `public async Task SetWords(List<Word> words)` does not exist anymore.

Commit:
Replaced DefaultData with JSON File

DefaultData now uses a JSON file with over 5000 words in the "Word"-type

MibFromGit committed:
Mainwindow Style

angefangen die Hauptseite zu stylen und Styles für Buttons erstellt

Z1Epime committed:
Changed default word procedure

The default words are no longer hardcoded in c#, but copied to the programm installation directory as .json instead.
The default words are no longer saved in the roaming folder in appdata.
